### ISSUE TITLE
Unable to start after full disk

### ISSUE THREAD

**Issue by Dynom** [coreos/etcd#4220]
It's related to #3591, but that ticket didn't help me, at all. Etcd2 is refusing to start and it's seriously missing information to figure out what to do next. It's highly frustrating.

I've a cluster of 5 machines. Due to a bug in some software the disk on one node got full. Now Etcd fails to start, claiming a file can't be found, no idea what file it tries to locate.

* In `snap/` there were several files ending on `.snap.broken`. All were 0 bytes large. I renamed them to .snap, but after a start of Etcd2, they got renamed back to `.snap.broken`. I removed them.
* In `member/` there were several files ending on `.wal` and one ending on `.wal.broken`. I stopped Etcd2, renamed the file to `.wal` and res-started Etcd2. The result is in the logs:

The log:
```
$ etcd2 --debug
2016-01-15 12:36:05.944772 I | flags: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://172.31.34.46:2379
2016-01-15 12:36:05.944840 I | flags: recognized and used environment variable ETCD_DATA_DIR=/var/lib/etcd2
2016-01-15 12:36:05.944861 I | flags: recognized and used environment variable ETCD_DISCOVERY=https://discovery.etcd.io/ff...xx
2016-01-15 12:36:05.944885 I | flags: recognized and used environment variable ETCD_ELECTION_TIMEOUT=1200
2016-01-15 12:36:05.944913 I | flags: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=http://172.31.34.46:2380
2016-01-15 12:36:05.944942 I | flags: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379
2016-01-15 12:36:05.944963 I | flags: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://172.31.34.46:2380
2016-01-15 12:36:05.944990 I | flags: recognized and used environment variable ETCD_NAME=39d22783e74d48528fdfe346314733a2
2016-01-15 12:36:05.945028 I | flags: recognized and used environment variable ETCD_PROXY=on
2016-01-15 12:36:05.945115 I | etcdmain: etcd Version: 2.2.0
2016-01-15 12:36:05.945135 I | etcdmain: Git SHA: e4561dd
2016-01-15 12:36:05.945148 I | etcdmain: Go Version: go1.4.2
2016-01-15 12:36:05.945161 I | etcdmain: Go OS/Arch: linux/amd64
2016-01-15 12:36:05.945175 I | etcdmain: setting maximum number of CPUs to 1, total number of available CPUs is 2
2016-01-15 12:36:05.945238 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2016-01-15 12:36:05.945310 I | etcdmain: listening for peers on http://172.31.34.46:2380
2016-01-15 12:36:05.945364 I | etcdmain: listening for client requests on http://0.0.0.0:2379
2016-01-15 12:36:05.945606 W | etcdserver: discovery token ignored since a cluster has already been initialized. Valid log found at "/var/lib/etcd2/member/wal"
2016-01-15 12:36:05.945653 I | etcdserver: name = 39d22783e74d48528fdfe346314733a2
2016-01-15 12:36:05.945687 I | etcdserver: data dir = /var/lib/etcd2
2016-01-15 12:36:05.945702 I | etcdserver: member dir = /var/lib/etcd2/member
2016-01-15 12:36:05.945716 I | etcdserver: heartbeat = 100ms
2016-01-15 12:36:05.945729 I | etcdserver: election = 1200ms
2016-01-15 12:36:05.945741 I | etcdserver: snapshot count = 10000
2016-01-15 12:36:05.945753 I | etcdserver: discovery URL= https://discovery.etcd.io/ff...xx
2016-01-15 12:36:05.945770 I | etcdserver: advertise client URLs = http://172.31.34.46:2379
2016-01-15 12:36:05.945881 C | etcdserver: open wal error: wal: file not found
```

No idea what file Etcd2 tries to load, all files I have are in the directory:
```
# ls -halF /var/lib/etcd2/member/wal
total 348M
drwx------ 2 etcd etcd 4.0K Jan 15 12:36 ./
drwx------ 4 etcd etcd 4.0K Jan 15 12:36 ../
-rw-r--r-- 1 etcd etcd  42M Oct 29 13:11 000000000000006f-0000000001730472.wal
-rw------- 1 etcd etcd  62M Jan 14 23:02 0000000000000291-00000000087c630f.wal
-rw------- 1 etcd etcd  62M Jan 15 02:26 0000000000000292-00000000087fb746.wal
-rw------- 1 etcd etcd  62M Jan 15 05:50 0000000000000293-0000000008830b7e.wal
-rw------- 1 etcd etcd  62M Jan 15 09:14 0000000000000294-0000000008865f60.wal
-rw------- 1 etcd etcd  33M Jan 15 11:06 0000000000000295-000000000889b34e.wal
```

Oddly, it's the one from Oct 29 that was the broken file:
```
-rw-r--r-- 1 etcd etcd  42M Oct 29 13:11 000000000000006f-0000000001730472.wal.broken
```

My highest priority is to get my cluster back in a running state. So my question is: **What to do next?** Preferably I re-introduce this node back to the cluster, otherwise I have to copy lots of data to a new node. In theory it can be done, however.

I'm currently inclined to start the steps on: https://coreos.com/etcd/docs/latest/runtime-configuration.html#replace-a-failed-machine but I feel that should be too heavy for my situation.

**Comment by xiang90** [comment]
@Dynom can you do a ls on you entire member data dir and paste the info here?

**Comment by Dynom** [comment]
Hi @xiang90,

certainly:
```
ls -halF
total 20K
drwxr-xr-x  3 etcd etcd 4.0K Jan 15 15:56 ./
drwxr-xr-x 22 root root 4.0K Jan 15 00:00 ../
drwx------  4 etcd etcd 4.0K Jan 15 15:56 member/
```

**Comment by Dynom** [comment]
I've already worked around the problem by migrating to a new node, so I
* Created a new instance
* Copied the data
* Added it as a member to the cluster
* Deleted the broken member from the cluster

**I've a question remaining:**
Once I empty the etcd2 directory, can safely re-add it as a new member on the same IP address? It was a member according to the original discovery URL (on discovery.etcd.io).

**Comment by xiang90** [comment]
> And a tree form:

Did you remove the entire snap dir?

**Comment by xiang90** [comment]
> I've already worked around the problem by migrating to a new node, so I

You do not have to do step 2. 

> Once I empty the etcd2 directory, can safely re-add it as a new member on the same IP address? It was a member according to the original discovery URL (on discovery.etcd.io).

You can do it. Discovery is only used for bootstrapping. After that, the entire cluster is in your control.

See https://github.com/coreos/etcd/blob/master/Documentation/runtime-reconf-design.md#do-not-use-public-discovery-service-for-runtime-reconfiguration.

**Comment by Dynom** [comment]
> Did you remove the entire snap dir?

No it's there, it just doesn't show up with the command arguments I used (`-type f`). Etcd2 refuses to start without that directory.

> You do not have to do step 2.

I wasn't talking about Etcd2 data, I was talking about the data on the server that I needed. I referred to that in my initial post.

> You can do it. Discovery is only used for bootstrapping. After that, the entire cluster is in your control.

Allright. I thought I read somewhere to be careful when re-adding nodes with the same IP. Perhaps that was outdated or I'm simply mistaken.
